src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -580,6 +580,22 @@ void primitive_inst::realloc_if_needed(bool prev_execution_skipped) {
     const auto& actual_layouts = updated_params.output_layouts;
     OPENVINO_ASSERT(actual_layouts[0].is_static(), "[GPU] Can't realloc mem for dynamic layout");
 
+    if (users.size() == 1 && users.front()->get_node().is_type<reorder>() && users.front()->can_be_optimized()) {
+        auto reorder_inst = users.front();
+        if (reorder_inst->is_output()
+            && reorder_inst->output_memory_ptr()
+            && get_network().has_output_remote_memory_ptr(reorder_inst->id())
+            && get_network().get_engine().is_the_same_buffer(get_network().get_output_remote_memory(reorder_inst->id()), reorder_inst->output_memory())) {
+            if (actual_layouts[0] == _impl_params->get_output_layout()) {
```

Comment:
Do we need this condition? Also if it is needed, i.e., if there is a case where actual layout != impl_params->output_layout, then shoudln't we do nothing?

Reply:
Yes, In most cases this condition may not be needed. It's just to separate the case when the current node is fc and actual laytout differs from original layout by get_fake_aligned_params

Reply:
BTW I think even though actual layout (fake aligned layout ) is not same as the impl_params->output_layout, 
we can still use remote tensor as the output memory of cur node
if the remote tensor size is equal or larger than the fake aligned layout.

Reply:
Also if we decided not to set the current output as reorder's remote tensor, then we need to set the reorder not to be optimized, arent we?

Reply:
Agree, Changed it so that if faked aligned layout is smaller than remote tensor size, it uses remote tensor, and if not, it cancels reorder opitimization